lib/collection/map.js

```diff
@@ -38,8 +38,11 @@ export class MapColl extends KeyValueCollection {
   }
 
   clear() {
-    this._notifyRemoved(this.contents);
-    this._map.clear();
+    if (this._map.size) {
+      let items = this.contents;
+      this._map.clear();
+      this._notifyRemoved(items);
+    }
```

Comment:
Make such things an early return. The normal code path should be straight, not indented.

```suggestion
    if (this.isEmpty) {
      return;
    }
    let items = this.contents;
    this._map.clear();
    this._notifyRemoved(items);
```